fleet_adapter_mir/fleet_adapter_mir/robot_adapter_mir.py

```diff
@@ -0,0 +1,767 @@
+                    f'[{self.name}] MissionHandle has changed to '
+                    f'{mission_handle.execution.identifier}, interrupting '
+                    f'navigation mission {navigation_mission_identifier}. '
+                    f'Stopping original navigation mission loop.')
+                break
+
+            try:
+                if destination.name:
+                    self.node.get_logger().info(
+                        f'[{self.name}] is going to MiR position '
+                        f'{destination.name}')
+                    mission_queue_id = self.api.go_to_known_position(
+                        destination.name)
+                if mission_queue_id is None:
+                    self.node.get_logger().info(
+                        f'[{self.name}] is going to MiR coordinates '
+                        f'[{destination.position}]')
+                    mission_queue_id = self.api.navigate(destination.position)
+                else:
+                    self.node.get_logger().info(
+                        f'[{self.name}] Move mission cannot be queued '
+                        f'(no mission queue id provided), retrying...')
+            except Exception as err:
+                self.node.get_logger().info(
+                    f'[{self.name}] Failed to request robot to move to '
+                    f'destination: {err}. Retrying... ({count})'
+                )
+            count += 1
+            time.sleep(1)
+
+        if (mission_queue_id is None and
+                navigation_mission_identifier !=
+                mission_handle.execution.identifier):
+            self.node.get_logger().info(
+                f'[{self.name}] Failed to request robot to move to '
+                f'destination. Navigation step '
+                f'{navigation_mission_identifier} interrupted by new mission '
+                f'{mission_handle.execution.identifier}.'
+            )
+            return
+
+        if mission_queue_id is None:
+            self.node.get_logger().info(
+                f'[{self.name}] Failed to request robot to move to '
+                f'destination. Maximum request navigate retries exceeded!')
+            mission_handle.execution.finished()
+            return
+
+        mission_handle.set_mission_queue_id(mission_queue_id)
+
+    @parallel
+    def request_go_to_known_position(
+        self,
+        position_name: str,
+        mission_handle: MissionHandle
+    ):
+        # Used for exiting while loop early in the event that for whatever
+        # reason the robot starts performing a different mission, while the
+        # original navigation mission is ongoing
+        navigation_mission_identifier = mission_handle.execution.identifier
+        mission_queue_id = None
+        count = 0
+        retry_count = 10
+        while count < retry_count and not mission_queue_id:
+            if (navigation_mission_identifier !=
+                    mission_handle.execution.identifier):
```

Comment:
You should use [`mission_handle.activity`](https://github.com/open-rmf/fleet_adapter_mir/blob/fc2e76993df4cffb2bc4b7e3555fc2aa2380750c/fleet_adapter_mir/fleet_adapter_mir/robot_adapter_mir.py#L62) here, just in case `mission_handle.execution` ever gets replaced with a `None` value.